src/pydicom/sequence.py

```diff
@@ -51,17 +35,12 @@ def __init__(self, iterable: Iterable[Dataset] | None = None) -> None:
         if isinstance(iterable, Dataset):
             raise TypeError("The Sequence constructor requires an iterable")
 
-        # validate_dataset is used as a pseudo type_constructor
-        self._list: list[Dataset] = []
-        # If no inputs are provided, we create an empty Sequence
-        super().__init__(validate_dataset, iterable or [])
+        # If True, SQ element uses an undefined length of 0xFFFFFFFF
         self.is_undefined_length: bool
```

Comment:
I wonder where we need this. It is set during reading, but I couldn't find where it is read... And if it is really needed, maybe it should by an argument in `__init__`?
(not related to the changes, just a random thought)

Reply:
Hmm, you're right. It doesn't seem to be used by anything